genai-perf/genai_perf/parser.py

```diff
@@ -244,6 +244,24 @@ def _check_load_manager_args(args: argparse.Namespace) -> argparse.Namespace:
         args.concurrency = 1
     return args
 
+def _check_goodput_args(args):
```

Comment:
Please look at how extra inputs works and see if you can use the same code (or even refactor out a function). Goodput constraints should be turned into a dictionary where the users can put in any value of --goodput-constraint <key:value> --goodput-constraint <key:value> and so on, where order does not matter. Those keys should match any statistic available for that model. The value should be able to be a float or an int. This is the biggest change I'd like to see.

If the model has a statistic available that matches that key, then it should be used. If not, then goodput can be set to N/A and we can print out a message stating that goodput could not be calculated because <key> does not exist as a statistic for the model.